lib/pengine/status.c

```diff
@@ -98,23 +99,25 @@ check_for_deprecated_rules(pcmk_scheduler_t *scheduler)
  *  - A list of nodes that need to be stonith'd
  *  - A list of nodes that need to be shutdown
  *  - A list of the possible stop/start actions (without dependencies)
+ *
+ * \return Standard Pacemaker return code
  */
-gboolean
-cluster_status(pcmk_scheduler_t * scheduler)
+int
+pcmk_unpack_scheduler_input(pcmk_scheduler_t *scheduler)
```

Comment:
I don't even know if we want to continue exposing this or not. It might be better to rely on higher-level functions, and make sure we have higher-level functions that do what sbd needs.

The way I'm thinking about it, nothing external should care about the internal state of a `pcmk_scheduler_t` object. External code like sbd needs to get certain information, which it can get by calling higher-level functions like `pcmk_has_quorum()`, `pcmk_find_node()`, etc. Those higher-level functions could start making sure the input (if non-NULL) has been unpacked before doing their work.

Reply:
@nrwahl2 Do you think it's worth continuing with this PR?